### ISSUE TITLE
cloudflare_logpush_ownership_challenge does not work with cloudflare_logpush_job

### ISSUE THREAD

**Issue by gpacuilla** [cloudflare/terraform-provider-cloudflare#1191]
### Confirmation

- [X] My issue isn't already found on the issue tracker.
- [X] I have replicated my issue using the latest version of the provider and it is still present.

### Terraform and Cloudflare provider version

Terraform v1.0.3
on darwin_amd64
+ provider registry.terraform.io/cloudflare/cloudflare v2.26.1
+ provider registry.terraform.io/hashicorp/google v3.83.0

### Affected resource(s)

* cloudflare_logpush_ownership_challenge

### Terraform configuration files

```hcl
resource "cloudflare_logpush_ownership_challenge" "zone" {
  zone_id          = cloudflare_zone.zone.id
  destination_conf = "${var.logpush_bucket}/${var.zone}"
}

data "google_storage_bucket_object_content" "zone_ownership_challenge" {
  name   = cloudflare_logpush_ownership_challenge.zone.ownership_challenge_filename
  bucket = split("gs://", var.logpush_bucket)[1]
}

resource "cloudflare_logpush_job" "zone" {
  enabled             = true
  zone_id             = cloudflare_zone.zone.id
  dataset             = "http_requests"
  logpull_options     = var.logpush_options
  destination_conf    = "${var.logpush_bucket}/${var.zone}/{DATE}"
  ownership_challenge = data.google_storage_bucket_object_content.zone_ownership_challenge.content
}
```

### Debug output

N/A

### Panic output

N/A

### Expected output

Expect the `cloudflare_logpush_job` resource to be created succesfully.

### Actual output

```
Error: error creating logpush job: HTTP status 400: incorrect ownership challenge (1002)
```

### Steps to reproduce

With the included terraform, the `cloudflare_logpush_ownership_challenge` works as expected and we can see the ownership-challenge file newly created in the bucket.

The `google_storage_bucket_object_content` data source gets the proper contents of the ownership challenge (double-checked manually) and is passed properly into the logpush job resource as seen in the diff.

However, when trying to create the `cloudflare_logpush_job` resource it errors out with HTTP/400 `incorrect ownership challenge (1002)`.

**Comment by gpacuilla** [comment]
Oops - maybe related to #1019

**Comment by jacobbednarz** [comment]
Please provide the TF_LOG=DEBUG output as the template has requested.

**Comment by gpacuilla** [comment]
I can confirm this was related to the mismatched `destination_conf` on the ownership_challenge and the job resources. Including the `{DATE}` subdirectory on both fixed the issue as also mentioned in that #1019 - sorry I missed that one before posting this.

However, have a new issue which requires running terraform twice to fully apply the results:

```
 Error: Provider produced inconsistent final plan

 When expanding the plan for cloudflare_logpush_job.zone to include new values learned so far during apply, provider "registry.terraform.io/cloudflare/cloudflare" produced an invalid new
 value for .ownership_challenge: was null, but now
 cty.StringVal("long-ownership-challenge-string").

 This is a bug in the provider, which should be reported in the provider's own issue tracker.
```

This is when using the (rough) code example above - how could we get the `cloudflare_logpush_job` resource to properly wait on the `google_storage_bucket_object_content` data source? Currently, we just need to apply terraform twice in order for it to fully apply, which is still much better than before (manually creating and then importing the job)

I've tried adding `depends_on` to the cloudflare_log_push resource on the GCS object content data but that didn't help and still caused the same `invalid value` error.

**Comment by jacobbednarz** [comment]
I'm afraid we don't really have a good solution for that as it depends on the object existing in GCS which the resource doesn't necessarily confirm. Instead, I would recommend applying it in two parts (challenge and then logpush config) to ensure it's done in order. Alternatively, use something like the `local-exec` provider to check the GCS file existence as a `depends_on` for the logpush config to exist.

**Comment by jacobbednarz** [comment]
Horrible but something else that may work is using the `time` provider to sleep N seconds before continuing - https://registry.terraform.io/providers/hashicorp/time/latest/docs/resources/sleep

**Comment by mveroone** [comment]
For the record, the `time` method described above doesn't work.
